src/depthai_wrapper.cpp

```diff
@@ -0,0 +1,133 @@
+        }
+    } else {
+        rgb_height_ = 3040;
+        rgb_width_ = 4056;
+        mono_height_ = 720;
+        mono_width_ = 1280;
+    }
+}
+
+void DepthAI::create_frame_holders()
+{
+    // creating place holders for all the streams enabled in config
+    if (config_json_.contains("streams")) {
+        for (const auto& it : config_json_.at("streams")) {
+            if (it.is_string()) {
+                if (it.get<std::string>() == "metaout")
+                    continue;
+                if (it.get<std::string>() == "previewout") {
+                    CV_mat_ptr img = std::make_shared<cv::Mat>(rgb_height_, rgb_width_, CV_8UC3);
+                    image_stream_holder["previewout"] = img;
+                }
+            } else {
+                const auto& name = it.at("name").get<std::string>();
+                if (name == "previewout") {
+                    CV_mat_ptr img = std::make_shared<cv::Mat>(rgb_height_, rgb_width_, CV_8UC3);
+                    image_stream_holder["previewout"] = img;
+                } else if (name == "depth") {
+                    CV_mat_ptr img = std::make_shared<cv::Mat>(mono_height_, mono_width_, CV_16UC1);
+                    image_stream_holder["depth"] = img;
+                } else if (name == "disparity_color") {
+                    CV_mat_ptr img = std::make_shared<cv::Mat>(mono_height_, mono_width_, CV_16UC3);
+                    image_stream_holder["disparity_color"] = img;
+                }
+            }
+        }
+    }
+}
+
+// How about creating streams for everything in the vector in constructor and then updating them
+// each time get frames is called ? How will it receive if it is not called for a small span of time ?
+
+// TODO(sachin): Crash on size mismatch. DO this later.
+// TODO(sachin): modify it using a struct to include time stamps and not loose frames.
+
+void DepthAI::get_frames(std::unordered_map<std::string, CV_mat_ptr>& output_streams)
+{
+    int count = image_stream_holder.size();
+    std::set<std::string> dirty_check;
+    while (count) { // count and dirty check is used incase same stream appears twice before other streams.
+        packets_ = pipeline_->getAvailableNNetAndDataPackets(true);
+
+        // iterating over the packets to extract all the image streams specified in the config
+        for (const auto& sub_packet : std::get<1>(packets_)) {
+            std::unordered_map<std::string, CV_mat_ptr>::iterator it = image_stream_holder.find(sub_packet->stream_name);
+            if (it != image_stream_holder.end()) {
+                unsigned char* img_ptr = reinterpret_cast<unsigned char*>((it->second)->data);
+                const auto& received_data = sub_packet->getData();
+
+                memcpy(img_ptr, received_data, sub_packet->size());
+                if (dirty_check.find(sub_packet->stream_name) == dirty_check.end()) {
+                    dirty_check.insert(sub_packet->stream_name);
+                    count--;
```

Comment:
So here's how I understand this. Tell me if I'm wrong. 
1. You want `output_streams` to have `count` streams. So you are going to keep receiving stuff until `count = 0`.
2. If there's a repeat, this loop will actually update the stream data, but not reduce `count` since it is a repeat.
Right?

Reply:
-did they way you asked but autoformatting keeps changing it back